modules/core/src/main/java/org/apache/ignite/internal/processors/performancestatistics/FilePerformanceStatisticsWriter.java

```diff
@@ -205,11 +218,10 @@ public void transaction(GridIntList cacheIds, long startTime, long duration, boo
      * @param success Success flag.
      */
     public void query(GridCacheQueryType type, String text, long id, long startTime, long duration, boolean success) {
```

Comment:
Cant `text` be null?

Reply:
I have checked all places, it can't. Moreover in-memory metrics have assertion `assert qry != null;` (QueryHistoryKey)

Reply:
What about SCAN query? What text contains those queries?

Reply:
In case of SCAN query it represents cache name. It's also can't be null.